src/main/java/spoon/support/JavaOutputProcessor.java

```diff
@@ -96,7 +122,7 @@ public void init() {
 			}
 		}
 		try {
-			directory = directory.getCanonicalFile();
+			this.getEnvironment().setSourceOutputDirectory(directory.getCanonicalFile());
```

Comment:
I'm not sure about this one: this should be done directly in the setter of Environment, no? WDYT?